src/Analysis/Ast/Test/ImportTests.cs

```diff
@@ -234,5 +238,57 @@ import sys
+
+        [TestMethod, Priority(0)]
+        public async Task ModuleInternalImportSys() {
+            var appUri = TestData.GetTestSpecificUri("app.py");
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("package", "__init__.py"), "from . import m1\nimport sys");
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("package", "m1", "__init__.py"), string.Empty);
+
+            await CreateServicesAsync(PythonVersions.LatestAvailable3X);
+            var rdt = Services.GetService<IRunningDocumentTable>();
+            var doc = rdt.OpenDocument(appUri, "import package");
+
+            await Services.GetService<IPythonAnalyzer>().WaitForCompleteAnalysisAsync();
+            var analysis = await doc.GetAnalysisAsync(Timeout.Infinite);
+            analysis.Should().HaveVariable("package").Which.Should().HaveMembers("m1", "sys");
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task ModuleImportingSubmodule() {
+            var appUri = TestData.GetTestSpecificUri("app.py");
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("package", "__init__.py"), "import package.m1");
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("package", "m1", "__init__.py"), string.Empty);
+
+            await CreateServicesAsync(PythonVersions.LatestAvailable3X);
+            var rdt = Services.GetService<IRunningDocumentTable>();
+            var doc = rdt.OpenDocument(appUri, "import package");
+
+            await Services.GetService<IPythonAnalyzer>().WaitForCompleteAnalysisAsync();
+            var analysis = await doc.GetAnalysisAsync(Timeout.Infinite);
+            analysis.Should().HaveVariable("package").Which.Should().HaveMember("m1");
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task ModuleImportingSubmodules() {
+            var appUri = TestData.GetTestSpecificUri("app.py");
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("top", "__init__.py"), @"
+from top import sub1
+import top.sub2
+import top.sub3.sub4
+");
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("top", "sub1.py"), string.Empty);
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("top", "sub2.py"), string.Empty);
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("top", "sub3", "__init__.py"), string.Empty);
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("top", "sub3", "sub4.py"), string.Empty);
+
+            await CreateServicesAsync(PythonVersions.LatestAvailable3X);
+            var rdt = Services.GetService<IRunningDocumentTable>();
+            var doc = rdt.OpenDocument(appUri, "import top");
+
+            await Services.GetService<IPythonAnalyzer>().WaitForCompleteAnalysisAsync();
+            var analysis = await doc.GetAnalysisAsync(Timeout.Infinite);
+            analysis.Should().HaveVariable("top").Which.Should().HaveMembers("sub1", "sub2", "sub3", "top");
```

Comment:
What should happen here in `__init__.py` itself, e.g.:
```cs
var top = rdt.GetDocument(TestData.GetTestSpecificUri("top", "__init__.py"));
var topAnalysis = await top.GetAnalysisAsync(-1);
analysis.Should().HaveVariable("sub2").And.NotHaveVariable("sub3");
```

Reply:
Also, I would've check `sub3` with something like this:
```cs
await TestData.CreateTestSpecificFileAsync(Path.Combine("top", "sub3", "__init__.py"), @"
import top.sub3.sub4
");
```
and then
```cs
var sub3 = rdt.GetDocument(TestData.GetTestSpecificUri("top", "__init__.py"));
var sub3Analysis = await sub3.GetAnalysisAsync(-1);
analysis.Should().HaveVariable("sub4").And.NotHaveVariable("sub3");
```

Reply:
`analysis.Should().HaveVariable("sub2").And.NotHaveVariable("sub3");` - `top` actually has `sub3` which is produced by `import top.sub3.sub4`, as tested in the console.

Reply:
And `import top.sub3.sub4` does not seem to declared `sub4`

Reply:
I.e. in the 
```
Python 3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 23:09:28) [MSC v.1916 64 bit (AMD64)] on win32
>>> import top
>>> dir(top)
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', 'sub1', 'sub2', 'sub3', 'top']
>>>
```
So basically
```python
from top import sub1
import top.sub2
import top.sub3.sub4
```
yields members `top`, `sub1`, `sub2` and `sub3` but not `sub4`.